conftest.py

```diff
@@ -11,20 +11,72 @@
 ###############################################################################
 
 import sys
+import os
+import logging
+
+import pytest
+from typing import Mapping
+from hashlib import md5
 
 from spinalcordtoolbox.utils import sct_dir_local_path, sct_test_path
 sys.path.append(sct_dir_local_path('scripts'))
 
-import pytest
 import sct_download_data as downloader
 
 
-@pytest.fixture(scope='session', autouse=True)
-def download_data(request):
-    # This is a hack because the capsys fixture can't be used with
-    # session scope at the moment.
-    # https://github.com/pytest-dev/pytest/issues/2704#issuecomment-603387680
-    capmanager = request.config.pluginmanager.getplugin("capturemanager")
-    with capmanager.global_and_fixture_disabled():
-        print('\nDownloading sct testing data.')
-        downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+logger = logging.getLogger(__name__)
+
+
+def pytest_sessionstart():
+    """ Download sct_testing_data prior to test collection. """
+    logger.info("Downloading sct test data")
+    downloader.main(['-d', 'sct_testing_data', '-o', sct_test_path()])
+
+
+@pytest.fixture(scope="session", autouse=True)
+def test_data_integrity(request):
+    files_checksums = dict()
+    for root, _, files in os.walk(sct_test_path()):
+        for f in files:
+            fname = os.path.join(root, f)
+            chksum = checksum(fname)
+            files_checksums[fname] = chksum
```

Comment:
I have one idea for a way to refactor this, but I'm not sure it's too critical.

Could this be its own function, e.g. `generate_checksums()` that returns `files_checksums`?

That way, `check_testing_data_integrity` could just compare `files_checksums_before` and `files_checksums_after` e.g. https://stackoverflow.com/a/18860653. Then the directory walking code wouldn't be repeated, too.

Reply:
You're technically right. My instinct would be to leave it for now since it's a minor cosmetic change and revisit it later if there is a need to compute checksums somewhere else in the project (for the enhanced `downloader` perhaps?). We'll see if/when this happens.

You're welcome to refactor it though.

Reply:
That makes sense to me. :)